pkg/krmtotf/references_test.go

```diff
@@ -642,6 +647,102 @@ func TestResolveResourceReferenceToTFResource(t *testing.T) {
 	}
 }
 
+func TestResolveResourceReferenceToTFResource_deleting(t *testing.T) {
+	testId := testvariable.NewUniqueId()
+	c := mgr.GetClient()
+	gvk := schema.GroupVersionKind{Group: "test1.cnrm.cloud.google.com", Version: "v1alpha1", Kind: "Foo"}
+	ns := testId
+	testcontroller.EnsureNamespaceExistsT(t, c, ns)
+	resource := &Resource{}
+	resource.SetGroupVersionKind(gvk)
+	resource.SetNamespace(ns)
+	resource.SetDeletionTimestamp(&metav1.Time{Time: time.Now()})
+	tfField := "bar_field"
+	tests := []struct {
+		name                string
+		config              map[string]interface{}
+		referencedResources []*unstructured.Unstructured
+		refConfig           v1alpha1.ReferenceConfig
+		expectedFinalConfig map[string]interface{}
+		shouldError         bool
```

Comment:
I think I still want to keep `shouldError` here, the goal of the test is to make sure no error under this scenario, `shouldError: true` highlights that. Otherwise I feel this test might be confusing.

Reply:
Reading it together with the comment below, it makes sense.